### PR TITLE
Enable linter revive and resolve found errors: part1

### PR DESCRIPTION
**Pull request by blackpiglet**
Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [x] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: pkg/install/resources.go [hunk 1]
```diff
@@ -293,31 +293,31 @@ func AllResources(o *VeleroOptions) *unstructured.UnstructuredList {
...
 	deployOpts := []podTemplateOption{
-		WithAnnotations(o.PodAnnotations),
-		WithLabels(o.PodLabels),
-		WithImage(o.Image),
-		WithResources(o.VeleroPodResources),
-		WithSecret(secretPresent),
-		WithDefaultRepoMaintenanceFrequency(o.DefaultRepoMaintenanceFrequency),
-		WithServiceAccountName(serviceAccountName),
-		WithGarbageCollectionFrequency(o.GarbageCollectionFrequency),
-		WithUploaderType(o.UploaderType),
```
Comment (kaovilai) [inline pkg/install/resources.go:304]: We [use these functions](https://github.com/openshift/oadp-operator/blob/94cf689eee35e51d34804393f6291aca21d5a52a/controllers/velero.go#L311-L320) in openshift/oadp-operator. Any way to keep these exported?

File: pkg/install/resources.go [hunk 2]
```diff
@@ -326,15 +326,15 @@ func AllResources(o *VeleroOptions) *unstructured.UnstructuredList {
 
 	if o.UseNodeAgent {
 		dsOpts := []podTemplateOption{
-			WithAnnotations(o.PodAnnotations),
-			WithLabels(o.PodLabels),
-			WithImage(o.Image),
-			WithResources(o.NodeAgentPodResources),
-			WithSecret(secretPresent),
-			WithServiceAccountName(serviceAccountName),
```
Comment (kaovilai) [inline pkg/install/resources.go:334]: We [use these functions](https://github.com/openshift/oadp-operator/blob/94cf689eee35e51d34804393f6291aca21d5a52a/controllers/restic.go#L177-L182) in openshift/oadp-operator. Any way to keep these exported?

**Review by kaovilai — CHANGES_REQUESTED** [review]

File: pkg/install/resources.go (hunk 1 above)
Comment (blackpiglet) [reply pkg/install/resources.go:304]: Sure. I will make some change to ignore this linter warning.

File: pkg/install/resources.go (hunk 2 above)
Comment (blackpiglet) [reply pkg/install/resources.go:334]: Sure. I will make some change to ignore this linter warning.

**Review by Lyndon-Li — DISMISSED** [review]

File: pkg/install/resources.go (hunk 1 above)
Comment (blackpiglet) [reply pkg/install/resources.go:304]: unexported-return: exported func WithFeatures returns unexported type install.podTemplateOption, which can be annoying to use (revive)

**Review by kaovilai — APPROVED** [review]
These lgtm! Thanks!

**Review by Lyndon-Li — APPROVED** [review]

**Review by qiuming-best — APPROVED** [review]

**Pull request merged**